### PR TITLE
Use stable Next.js v13.2.0 next/font

### LINKED ISSUE #21330: [Feature Request]: Support the package `next/font` (the replacement of `@next/font`)
### Is your feature request related to a problem? Please describe

Since next.js v13.2 deprecate the `@next/font` package in favor of the integrated `next/font` export. The storybook next.js integration only support the `@next/font` one for now. Using `next/font` trigger an error

image

### Describe the solution you'd like

Storybook should handle the two packages.

### Describe alternatives you've considered

_No response_

### Are you able to assist to bring the feature to reality?

yes, I can

### Additional context

- https://github.com/vercel/next.js/pull/46159

### PR DESCRIPTION
**Pull request by alex-ahumada**
Closes https://github.com/storybookjs/storybook/issues/21330

## What I did

Migrated @next/font to stable next/font present in next.js v13.2.0.

## How to test

1. Add Storybook to a project using nextjs framework
2. Add next/font to preview.js (example using css variables)
```javascript
import { Roboto } from 'next/font/google'

...

const fontSans = Roboto({
  weight: ['400', '700'],
  style: ['normal', 'italic'],
  variable: '--font-roboto',
  subsets: ['latin'],
})

...

export const decorators = [
  (StoryFn) => (
      <div className={fontSans.variable}>
        <StoryFn />
      </div>
  ),
]
```

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

### CODE REVIEW

**Comment by valentinpalkovic** [comment]
Hi @alex-ahumada 

Thank you so much for your contribution!

I like what you have done! Though, we have to make the changes less disruptive. I am ok with changing docs and examples immediately from `@next/font` to `next/font`; but we still have to support both variants! (`@next/font` and `next/font`). Can you adjust your PR to not drop the support for `@next/font`?

**Comment by alex-ahumada** [comment]
Hi @valentinpalkovic .

No problem! Last commit should bring `@next/font` support back.

File: code/frameworks/nextjs/src/font/babel/index.test.ts [hunk 1]
```diff
@@ -22,9 +22,9 @@ const randomObj = {}
 it('should transform AST properly', () => {
```
Comment (valentinpalkovic) [inline code/frameworks/nextjs/src/font/babel/index.test.ts:22]: Could we introduce another test to test that `@next/font` imports are still working? :)

Comment (alex-ahumada) [reply code/frameworks/nextjs/src/font/babel/index.test.ts]: Done!

**Comment by alex-ahumada** [comment]
ci/circleci: check fails, there are some problems with usePathName and useSearchParams from next/navigation after next.js v13.2.x update. Used in [Navigation.stories.tsx](https://github.com/storybookjs/storybook/blob/next/code/frameworks/nextjs/template/stories_default-ts/Navigation.stories.tsx)

Related issue: https://github.com/vercel/next.js/issues/46360

**Comment by valentinpalkovic** [comment]
https://github.com/storybookjs/storybook/pull/21304 Got merged 3 hours ago. Please merge next into your branch one more time :)

**Comment by alex-ahumada** [comment]
It seems we also have a missing headRenderedAboveThisLevel property in src/routing/app-router-provider.tsx(81,11)

File: code/lib/core-common/src/utils/get-storybook-info.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ const findConfigFile = (prefix: string, configDir: string) => {
 
 const getConfigInfo = (packageJson: PackageJson, configDir?: string) => {
   let storybookConfigDir = configDir ?? '.storybook';
-  const storybookScript = packageJson.scripts?.['storybook'];
```
Comment (valentinpalkovic) [inline code/lib/core-common/src/utils/get-storybook-info.ts:101]: Is this change intended?

Comment (alex-ahumada) [reply code/lib/core-common/src/utils/get-storybook-info.ts:101]: Yes, but it shouldn't have been included in a merge, sorry for that.
If you prefer the original notation I can revert it.

File: code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ import {
   getFontAxes,
   getUrl,
   validateData,
-} from '@next/font/dist/google/utils';
+} from 'next/dist/compiled/@next/font/dist/google/utils';
```
[the remark this answers is not in the document]
Comment (valentinpalkovic) [reply code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts]: Ok. I have pushed a commit to fix this issue. Let's see, how the pipeline behaves :)

[the remark this answers is not in the document]
Comment (alex-ahumada) [reply code/frameworks/nextjs/src/font/webpack/loader/google/get-font-face-declarations.ts]: Perfect, nice fix!

**Review by valentinpalkovic — APPROVED** [review]

**Comment by valentinpalkovic** [comment]
Well done @alex-ahumada! 🎉

**Comment by alex-ahumada** [comment]
Thanks @valentinpalkovic ! Great teamwork! 😄

**Comment by Cayan** [comment]
I still see the import error when I import `next/font` instead of `@next/font`

Storybook version: 7.0.0-beta.61
Next.js version: 13.2.3

Error: ```(0 , next_font_google__WEBPACK_IMPORTED_MODULE_3__.Inter) is not a function```
Similar errors for local fonts.

**Comment by valentinpalkovic** [comment]
@Cayan can you provide a reproduction?

**Comment by Cayan** [comment]
I can create one later today @valentinpalkovic
